app/models/user.rb

```diff
@@ -150,6 +151,10 @@ def admin_self_deactivated?
     id.to_s == last_deactivated_by
   end
 
+  def at_least_one_communication_preference_selected
+    errors.add(:base, " At least one communication preference must be selected.") unless receive_email_notifications || receive_sms_notifications
```

Comment:
Adding errors to base is a symptom of an object that could be split out

Reply:
but the error msg isn't tied to any one of the two specific attributes (email and sms notifications), so wouldn't it be appropriate to add to the base object in this case? @compwron